fondant/component.py

```diff
@@ -38,33 +49,71 @@ def from_file(
             path: Path to the component spec file
         """
         component_spec = ComponentSpec.from_file(path)
-        return cls(component_spec)
+        args_dict = vars(cls._add_and_parse_args(component_spec))
+        user_arguments = cls.get_user_arguments(args_dict, component_spec)
+        metadata = args_dict.pop("metadata")
+        metadata = json.loads(metadata) if metadata else {}
+
+        return cls(
+            component_spec,
+            input_manifest_path=args_dict.pop("input_manifest_path"),
+            output_manifest_path=args_dict.pop("output_manifest_path"),
+            metadata=metadata,
+            user_arguments=user_arguments,
+        )
+
+    @classmethod
+    def from_args(cls) -> "Component":
+        """Create a component from a passed argument containing the specification as a dict."""
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--component_spec", type=json.loads)
+        args, _ = parser.parse_known_args()
+
+        if not args.component_spec:
+            raise ValueError("Error: The --component_spec argument is required.")
+
+        component_spec = ComponentSpec(args.component_spec)
```

Comment:
I think this is quite specific to initializing `Component` with the `from_args` class method. 

We're still using the the `from_file` constructor in the [pipeline script](https://github.com/ml6team/fondant/blob/43567274fc00c87c96011ec0dc3c18931d54e10a/fondant/pipeline.py#LL70C1-L70C1) where we read the spec directly from the path and not as a serialized argument.

Reply:
Do we actually want to pass in the spec directly instead of a path to it? I think passing the path will be easier for users, as I don't expect them to define the spec dynamically.

That would allow us to call `from_file()` from `from_args()` as now they contain a lot of duplication.

Reply:
The current flow looks like this: 

* Users pass the spec by referencing a local file path in the [`ComponentOp`](https://github.com/ml6team/fondant/blob/919394c85a1e262a213ce79c4adc9ac5c49c79ef/fondant/pipeline.py#LL53C9-L53C28)
* This gets passed as a dictionary argument to the component as a [`component_spec` argument](https://github.com/ml6team/fondant/blob/919394c85a1e262a213ce79c4adc9ac5c49c79ef/fondant/pipeline.py#LL355C33-L355C33) 
* The argument gets parsed at runtime in the `from_args()` method ([example](https://github.com/ml6team/fondant/blob/919394c85a1e262a213ce79c4adc9ac5c49c79ef/components/download_images/src/main.py#L154))

The user never has to pass the spec directly since we handle this internally. Passing a reference to a local file path would not work when you run the component remotely.

Reply:
Ok makes sense!

Can we then add a `from_spec()` method which contains everything below this line as this is currently duplicated in both `from_file()` and `from_args()`?

Reply:
Good suggestion :) 
Updated!